runners/tez/src/main/java/org/apache/beam/runners/tez/translation/TranslationContext.java

```diff
@@ -0,0 +1,149 @@
+package org.apache.beam.runners.tez.translation;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.apache.beam.runners.tez.TezPipelineOptions;
+import org.apache.beam.sdk.runners.AppliedPTransform;
+import org.apache.beam.sdk.runners.TransformHierarchy;
+import org.apache.beam.sdk.values.PValue;
+import org.apache.beam.sdk.values.TupleTag;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.hadoop.io.BytesWritable;
+import org.apache.tez.dag.api.DAG;
+import org.apache.tez.dag.api.DataSinkDescriptor;
+import org.apache.tez.dag.api.DataSourceDescriptor;
+import org.apache.tez.dag.api.TezConfiguration;
+import org.apache.tez.dag.api.Vertex;
+import org.apache.tez.dag.api.Edge;
+import org.apache.tez.runtime.library.conf.OrderedPartitionedKVEdgeConfig;
+import org.apache.tez.runtime.library.conf.UnorderedKVEdgeConfig;
+import org.apache.tez.runtime.library.partitioner.HashPartitioner;
+
+/**
+ * Maintains context data for {@link TransformTranslator}s.
+ * Tracks and maintains each individual {@link Vertex} and their {@link Edge} connections.
+ */
+public class TranslationContext {
```

Comment:
<!--new_thread; commit:346274d62540a16c9bf795fc6e9ab2cd28430008; resolved:0-->
Generally, runners are small enough pieces of software that it is nice to move everything into one Java package and just make the PipelineRunner and PipelineResult classes, and maybe a couple others, public. You can use package private classes to good effect.